projects/plugins/jetpack/extensions/shared/use-width/index.js

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useRef } from '@wordpress/element';
+import { WidthPanel } from './controls';
+
+const usePrevious = value => {
+	const ref = useRef();
+
+	useEffect( () => {
+		ref.current = value;
+	}, [ value ] );
+
+	return ref.current;
+};
+
+export default function useWidth( { attributes, setAttributes } ) {
+	const { align, width } = attributes;
```

Comment:
I wonder if it's correct to couple `align` and `width`. 

Do all blocks using `width` also use `align`? 🤔 
Or, like, is that a reasonable expectation?

Those are not rhetorical questions; I really don't know. 😄 
It kind of caught my eye because the PR description states:

> This PR abstracts the button width settings in a new useWidth hook so they can be used by any other block with a `width` attribute.

Whereas, in fact, it's:

> any block with a `width` and `align` attribute.

Previously, we used this logic only for the Button block, which has both.
Now that we are generalizing it, we might (or might not!) stumble upon blocks that only use `width` and would expect `useWidth` to work with them.

Reply:
> > Whereas, in fact, it's:
> any block with a `width` and `align` attribute.

I disagree 🙂 .

This is a side effect that changes the `width` attribute for blocks that have an `align` attribute. The hook doesn't require the block to have an `align` attribute (this side effect will simply not run since both `align` and `previousAlign` will be `undefined`).

Reply:
Makes perfect sense; thanks for looking into it! 🙇‍♂️

Reply:
I'd also say that the description of the PR made more sense when the hook also provided the `WidthPanel` component, but someone suggested to decouple it 😄